support-workers/src/main/scala/com/gu/support/workers/lambdas/CreatePaymentMethod.scala

```diff
@@ -106,42 +105,32 @@ class CreatePaymentMethod(servicesProvider: ServiceProvider = ServiceProvider)
       stripeService: StripeService,
       currency: Currency,
   ): Future[CreditCardReferenceTransaction] = {
-    val stripeServiceForCurrency = stripeService.withCurrency(currency)
-    stripe match {
-      case StripeSourcePaymentFields(source, stripePaymentType) =>
-        stripeServiceForCurrency.createCustomerFromToken(source).map { customer =>
-          val card = customer.source
-          CreditCardReferenceTransaction(
-            card.id,
-            customer.id,
-            card.last4,
-            CountryGroup.countryByCode(card.country),
-            card.exp_month,
-            card.exp_year,
-            card.brand.zuoraCreditCardType,
-            PaymentGateway = chargeGateway(currency),
-            StripePaymentType = stripePaymentType,
-          )
-        }
-      case StripePaymentMethodPaymentFields(paymentMethod, stripePaymentType) =>
-        for {
-          stripeCustomer <- stripeServiceForCurrency.createCustomerFromPaymentMethod(paymentMethod)
-          stripePaymentMethod <- stripeServiceForCurrency.getPaymentMethod(paymentMethod)
-        } yield {
-          val card = stripePaymentMethod.card
-          CreditCardReferenceTransaction(
-            paymentMethod.value,
-            stripeCustomer.id,
-            card.last4,
-            CountryGroup.countryByCode(card.country),
-            card.exp_month,
-            card.exp_year,
-            card.brand.zuoraCreditCardType,
-            PaymentGateway = paymentIntentGateway(currency),
-            StripePaymentType = stripePaymentType,
-          )
-        }
+    val stripeServiceForCurrency = {
+      stripe.stripePublicKey match {
+        case Some(stripePublicKey) => stripeService.withPublicKey(stripePublicKey)
+        case None => stripeService.withCurrency(currency)
```

Comment:
still temporarily uses the withCurrency method in case one of the checkouts isn't updated yet!